lib/cache/collections.go

```diff
@@ -2503,3 +2510,76 @@ func (accessListMembersExecutor) getReader(cache *Cache, cacheOK bool) services.
 }
 
 var _ executor[*accesslist.AccessListMember, services.AccessListMembersGetter] = accessListMembersExecutor{}
+
+type accessListReviewsExecutor struct{}
+
+func (accessListReviewsExecutor) getAll(ctx context.Context, cache *Cache, loadSecrets bool) ([]*accesslist.Review, error) {
+	// Get all access lists
+	var accessLists []*accesslist.AccessList
+	var nextToken string
+	for {
+		var page []*accesslist.AccessList
+		var err error
+
+		page, nextToken, err = cache.AccessLists.ListAccessLists(ctx, 0 /* default page size */, nextToken)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		accessLists = append(accessLists, page...)
+
+		if nextToken == "" {
+			break
+		}
+	}
+
+	// Get the reviews of each access list.
+	var reviews []*accesslist.Review
+	for _, accessList := range accessLists {
+		for {
+			var page []*accesslist.Review
+			var err error
+
+			page, nextToken, err = cache.AccessListReviews.ListAccessListReviews(ctx, accessList.GetName(), 0 /* default page size */, nextToken)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+
+			reviews = append(reviews, page...)
+
+			if nextToken == "" {
+				break
+			}
+		}
+	}
+	return reviews, nil
+}
+
+func (accessListReviewsExecutor) upsert(ctx context.Context, cache *Cache, resource *accesslist.Review) error {
+	// Reviews can't be updated, so we'll only support create.
```

Comment:
It's perfectly possible to have spurious OpPut events for the same write, so if this ends up attempting to create the same review twice we'll error out of the cache. For a resource to be in the cache, it needs an upsert operation.

Reply:
I've made the change and had this attempt to delete first. This behavior exists elsewhere in `collections.go`.